metametamodel/metametamodel.adoc

```diff
@@ -378,13 +378,74 @@ An EnumerationLiteral is similar to Ecore’s `EEnumLiteral` and to MPS’ `Enum
 .Characteristics
 An EnumerationLiteral has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
 
-An Enumeration is a <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
+An EnumerationLiteral is a <<IKeyed>> (as it has a key) and indirectly an <<INamed>> (as it has a name).
 
 .Constraints
 Each EnumerationLiteral must have a unique `name` within the Enumeration.
 
 Each EnumerationLiteral belongs to one and only one Enumeration.
 
+[[StructuredDatatype]]
+==== StructuredDatatype
+Represents a collection of named instances of <<PrimitiveType, Primitive Types>>.
+They are meant to support a small composite of values that semantically form a unit.
+StructuredDatatypes have no identity, are always copied by value, and SHOULD be immutable.{fn-org265}
+
+#TODO: use proper name for concept#
+
+.Example
+We could represent a "Complex number { real part, imaginary part }" or "Color { red, green, blue }" as StructuredDatatype.
+
+.EMF & MPS equivalent
+No direct similarities.
+We could simulate it by a custom `EDatatype` in ECore.
+
+.Characteristics
+[[StructuredDatatype.fields, StructuredDatatype.fields]]
+A StructuredDatatype contains <<Field, Fields>> in its `fields` containment.
+It also represents the <<namespaces, namespace>> for the fields.
+
+A StructuredDatatype has a <<INamed.name, `name`>> and an <<IKeyed.key, `key`>>.
+
+A StructuredDatatype is a <<DataType>> (as it can be used as <<Property.type, type>> of a <<Property>>).
+It is indirectly a <<LanguageEntity>> (as it is a top level element in a <<Language>>), an <<INamed>> (as it has a name) and a <<IKeyed>> (as it has a key).
+
+.Constraints
+M1 Instances of a StructuredDatatype SHOULD be immutable.
```

Comment:
Do we care if they are immutable or not? 
If they have no identity (i.e., we cannot refer to them) if someone modify a certain instance or create a new one , when passing them around the behavior remains the same, I think

Reply:
As far as I remember, we've decided to follow the rationale in #265: They SHOULD be immutable, so implementations can assume that and apply optimizations. It doesn't matter for the protocol itself.